src/modules/userMethods.ts

```diff
@@ -397,20 +404,48 @@ export const userMethods = {
         return prisma.user.findMany({ where: { memberships: { some: { groupId } }, active: { not: true } } });
     },
 
-    edit: async ({ id, savePreviousName, supplementalPosition, ...data }: EditUserFields) => {
-        if (data.organizationUnitId) {
+    edit: async ({
+        id,
+        savePreviousName,
+        supplementalPosition,
+        organizationUnitId,
+        curatorIds,
+        ...data
+    }: EditUserFields) => {
+        const updateUser: Prisma.UserUpdateInput = data;
+        const currentUser = await userMethods.getById(id);
+
+        if (organizationUnitId) {
             const newOrganization = await prisma.organizationUnit.findUnique({
                 where: {
-                    id: data.organizationUnitId,
+                    id: organizationUnitId,
                 },
             });
 
             if (!newOrganization) {
                 throw new TRPCError({
                     code: 'BAD_REQUEST',
-                    message: `No organization with id ${data.organizationUnitId}`,
+                    message: `No organization with id ${organizationUnitId}`,
                 });
             }
+
+            updateUser.organizationUnit = { connect: { id: organizationUnitId } };
+        }
+        if (curatorIds) {
+            const ids = curatorIds.map((id) => ({ id }));
+            const oldIds = currentUser.curators?.map(({ id }) => ({ id }));
+
+            const curatorIdsToConnect = calculateDiffBetweenArrays(ids, oldIds);
+            const curatorIdsToDisconnect = calculateDiffBetweenArrays(oldIds, ids);
+
+            updateUser.curators = {
+                connect: curatorIdsToConnect,
+                disconnect: curatorIdsToDisconnect,
+            };
+
+            if (curatorIds.includes(id)) {
```

Comment:
Again, this is not and id of current user, it's the edited user id
You should pass current user id separately